code/datums/components/overlay_lighting.dm

```diff
@@ -56,20 +60,35 @@
 	var/atom/movable/current_holder
 	///Movable atom the parent is attached to. For example, a flashlight into a helmet or gun. We'll need to track the thing the parent is attached to as if it were the parent itself.
 	var/atom/movable/parent_attached_to
-
-
-/datum/component/overlay_lighting/Initialize(_range, _power, _color, starts_on)
+	///Whether we're a directional light
+	var/directional
+	///Abstractional atom for directional light, we move this around to make the directional effect
+	var/obj/effect/abstract/directional_lighting/directional_atom
+	///A cone overlay for directional light, it's alpha and color are dependant on the light
+	var/obj/effect/overlay/light_cone/cone
+	///Current tracked direction for the directional cast behaviour
+	var/current_direction
+	///Cast range for the directional cast (how far away the atom is moved)
+	var/cast_range = 2
+
+/datum/component/overlay_lighting/Initialize(_range, _power, _color, starts_on, is_directional)
 	if(!ismovable(parent))
 		return COMPONENT_INCOMPATIBLE
 
 	var/atom/movable/movable_parent = parent
-	if(movable_parent.light_system != MOVABLE_LIGHT)
-		stack_trace("[type] added to [parent], with [movable_parent.light_system] value for the light_system var. Use [MOVABLE_LIGHT] instead.")
+	if(movable_parent.light_system != MOVABLE_LIGHT && movable_parent.light_system != MOVABLE_LIGHT_DIRECTIONAL)
+		stack_trace("[type] added to [parent], with [movable_parent.light_system] value for the light_system var. Use [MOVABLE_LIGHT] or [MOVABLE_LIGHT_DIRECTIONAL] instead.")
 		return COMPONENT_INCOMPATIBLE
 
 	. = ..()
 
 	visible_mask = new()
+	if(is_directional)
+		directional = TRUE
+		directional_atom = new()
+		cone = new()
+		cone.transform = cone.transform.Translate(-32, -32)
+		current_direction = SOUTH
```

Comment:
```suggestion
		set_direction(parent.dir)
```
Not sure if making two updates here will be an issue or not, but it should ideally not always start pointing towards `SOUTH`